prime-router/src/main/kotlin/fhirengine/engine/LookupTableValueSet.kt

```diff
@@ -40,8 +40,9 @@ class LookupTableValueSet
             throw SchemaException("Value column not found in specified lookup table")
         }
 
+        val filterTable = lookupTable.table.selectColumns(configData.keyColumn, configData.valueColumn)
```

Comment:
Originally, `retainColumns` was called directly on `lookupTable.table`, which was modifying the copy of the lookup table in metadata. Using `selectColumns` is more appropriate for filtering the table.